config/helpers.php

```diff
@@ -785,60 +786,11 @@ function tc(string $key, int $count)
  */
 function timestamp(?string $date = null, $step = null): ?int
 {
-    if (V::date($date) === false) {
-        return null;
-    }
-
-    $date = strtotime($date);
-
-    if ($step === null) {
-        return $date;
-    }
-
-    // fallback for pre-3.5.0 usage
-    if (is_int($step) === true) {
-        $step = [
-            'unit' => 'minute',
-            'size' => $step
-        ];
-    }
-
-    if (is_array($step) === false) {
-        return $date;
-    }
-
-    $parts = [
-        'second' => date('s', $date),
-        'minute' => date('i', $date),
-        'hour'   => date('H', $date),
-        'day'    => date('d', $date),
-        'month'  => date('m', $date),
-        'year'   => date('Y', $date),
-    ];
-
-    $current = $parts[$step['unit']];
-    $nearest = round($current / $step['size']) * $step['size'];
-    $parts[$step['unit']] = $nearest;
-
-    foreach ($parts as $part => $value) {
-        if ($part === $step['unit']) {
-            break;
-        }
-
-        $parts[$part] = 0;
-    }
-
-    $timestamp = strtotime(
-        $parts['year'] . '-' .
-        str_pad($parts['month'], 2, 0, STR_PAD_LEFT) . '-' .
-        str_pad($parts['day'], 2, 0, STR_PAD_LEFT) . ' ' .
-        str_pad($parts['hour'], 2, 0, STR_PAD_LEFT) . ':' .
-        str_pad($parts['minute'], 2, 0, STR_PAD_LEFT) . ':' .
-        str_pad($parts['second'], 2, 0, STR_PAD_LEFT)
-    );
+    $date = Date::optional($date);
+    $step = Date::stepConfig($step);
+    $date->round($step['unit'], $step['size']);
```

Comment:
A few potential breaking changes:
- doesn't check explicitly anymore if the string is a valid date
- always does rounding, event when no step is passed